File: strategy_pattern/go/duck/rubber_duck.go
package duck

import "duckapp/behavior"

type RubberDuck struct {
	Duck
}

func NewRubberDuck() *RubberDuck {
	rubber := &RubberDuck{}
	rubber.SetFlyBehavior(&behavior.FlyNoWay{})
	rubber.SetQuackBehavior(&behavior.Squeak{})
	return rubber
}

func (r *RubberDuck) Display() {
	println("I'm a rubber duckie")
}

File: strategy_pattern/ruby/ducks/rubber_duck.rb
require_relative '../behaviors/fly_behavior'
require_relative '../behaviors/quack_behavior'
require_relative 'duck'

class RubberDuck < Duck
  def initialize
    super
    @fly_behavior = FlyBehavior::FlyNoWay.new
    @quack_behavior = QuackBehavior::Squeak.new
  end

  def display
    puts "I'm a rubber duckie"
  end
end
